web_external/js/models/SessionModel.js

```diff
@@ -42,99 +36,54 @@ minerva.models.SessionModel = girder.models.ItemModel.extend({
+     */
+    createSessionMetadata: function () {
+        var metadata = this.metadata() || {};
+        // TODO this is highly map centric, probably should be split out
+        var map = {};
+        map.basemap = 'osm';
+        map.basemap_args = {
             tileUrl: 'https://{s}.tiles.mapbox.com/v3/datamade.hn83a654/{z}/{x}/{y}.png',
             attribution: '<a href=http://www.mapbox.com/about/maps/ target=_blank>Terms &amp; Feedback</a>'
         };
-        sessionJsonContents.center = {x: -100, y: 36.5};
-        sessionJsonContents.zoom = 4;
-        sessionJsonContents.features = [];
-        // now save this as session.json as a file in the item
-        this.sessionJsonFile = new girder.models.FileModel();
-        this.sessionJsonFile.on('g:upload.complete', function () {
-            callback();
-        }, this).uploadToItem(this, JSON.stringify(sessionJsonContents), 'session.json', 'application/json');
-    },
-
-    fetch: function () {
-        this.on('g:fetched', function () {
-            // now get the session file
-            var sessionParams = {
-                path: 'minerva_session/' + this.get('_id') + '/session',
-                type: 'GET'
-            };
-            girder.restRequest(sessionParams).done(_.bind(function (file) {
-                this.sessionJsonFile = new girder.models.FileModel(file);
-                // TODO if there isn't one need to create it
-                // now we have the file, get the actual contents
-                girder.restRequest({
-                    path: 'file/' + file._id + '/download',
-                    type: 'GET'
-                }).done(_.bind(function (resp) {
-                    this.sessionJsonContents = resp;
-                    this.trigger('m:fetched');
-                }, this)).error(_.bind(function (err) {
-                    console.error(err);
-                    girder.events.trigger('g:alert', {
-                        icon: 'cancel',
-                        text: 'Could not download session json contents.',
-                        type: 'error',
-                        timeout: 4000
-                    });
-                }, this));
-            }, this)).error(_.bind(function (err) {
-                console.error(err);
-                girder.events.trigger('g:alert', {
-                    icon: 'cancel',
-                    text: 'Could not load session item.',
-                    type: 'error',
-                    timeout: 4000
-                });
-            }, this));
-        }, this).on('g:error', function () {
-            minerva.router.navigate('sessions', {trigger: true});
-        }, this);
-        girder.models.ItemModel.prototype.fetch.call(this);
+        map.center = {x: -100, y: 36.5};
+        map.zoom = 4;
+        map.features = [];
+        metadata.map = map;
+        this.on('minerva.metadata.saved', function () {
+            this.trigger('minerva.session.saved', this);
```

Comment:
Is `minerva.` the new `m:`?

One problem with naming events like this is that jquery interprets them as namespaces which behave much like class selectors.  So this is really triggering a `minerva` event with namespaces `.session` and `.saved`.  It will trigger any of the following event handlers:
* `minerva`
* `.session`
* `.saved`
* `minerva.session`
* etc..

If you want to use namespaced events, then a better choice here would be `saved.minerva.metadata` (or `m:saved.minerva.metadata` keeping with girder tradition).